<!-- Purpose: launch waypoints generation node, turtlesim, differential drive model in rviz, fake_diff_encoders and odometer nodes for visualization-->
<launch>

<include file="$(find tsim)/launch/turtle_pent.launch"/>

<include file="$(find nuturtle_description)/launch/view_diff_drive.launch">
    <arg name="use_jsp_gui" value="false" doc="False for not using joint state publisher"/>
</include>

<node name="Obstacles" pkg="fake_laser_scan" type="obstacles">
      <rosparam file="$(find rigid2d)/config/params.yaml" />
</node>

<!-- Convert depth image to laser scan -->
<node name="FakeLaserScanner" pkg="fake_laser_scan" type="fake_laser_scan" output="screen" >
  <param name="use_sim_time" value="true"/>
  <rosparam file="$(find rigid2d)/config/params.yaml" />
</node>

<node name = "Odometer" pkg = "rigid2d"  type = "rigid2d_Odometer_node" output="screen" >
  <rosparam file="$(find rigid2d)/config/params.yaml" />
  <remap from="body_frame_id" to="~body_frame_id"/>
  <remap from="right_wheel_joint" to="~right_wheel_joint"/>
  <remap from="left_wheel_joint" to="~left_wheel_joint"/>
  <remap from="odom_frame_id" to="~odom_frame_id"/>
</node>

<node name = "Fake_Diff_Encoders" pkg = "rigid2d"  type = "rigid2d_Fake_Diff_Encoders_node" output="screen" >
  <rosparam file="$(find rigid2d)/config/params.yaml" />
  <remap from="body_frame_id" to="~body_frame_id"/>
  <remap from="right_wheel_joint" to="~right_wheel_joint"/>
  <remap from="left_wheel_joint" to="~left_wheel_joint"/>
  <remap from="odom_frame_id" to="~odom_frame_id"/>
  <remap from="/cmd_vel" to="/turtle1/cmd_vel"/>
</node>

<node name = "RealWorld" pkg = "rigid2d"  type = "rigid2d_Real_World_node" output="screen" >
  <rosparam file="$(find rigid2d)/config/params.yaml" />
  <remap from="body_frame_id" to="~body_frame_id"/>
  <remap from="right_wheel_joint" to="~right_wheel_joint"/>
  <remap from="left_wheel_joint" to="~left_wheel_joint"/>
  <remap from="map_frame_id" to="~map_frame_id"/>
  <remap from="slip_covariance" to="~slip_covariance"/>
  <remap from="odom_covariance" to="~odom_covariance"/>
  <remap from="odom_dift" to="~odom_dift"/>
</node>

</launch>
